### DISCUSSION TITLE
QUESTION - how to specify logger scope or AsyncLocal<> value?

### DISCUSSION THREAD

**Discussion by k0st1x** [quartznet/quartznet#2657] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
Need to create logger's scope (or AsyncLocal<>) to specify metadata for logic.

**Describe the solution you'd like**
i've tried
```cs
public class MyTriggerListener(ILogger logger) : ITriggerListener
{
  private IDisposable? scope;
  private AsyncLocal<string> someValue = new();
  public Task TriggerFired(ITrigger trigger, IJobExecutionContext context, CancellationToken cancellationToken)
  {
    scope = _logger.BeginScope("scope {value}", 42); // !!!
    someValue.Value = "42"; // !!!
    logger.LogInformation("someValue: ", someValue.Value); //  "scope 42" in logger scope
    return Task.CompletedTask;
  }
  public Task TriggerComplete(ITrigger trigger, IJobExecutionContext context, SchedulerInstruction triggerInstructionCode, CancellationToken cancellationToken)
  {
    scope?.Dispose();
    someValue.Value = null;
    return Task.CompletedTask;
  }
}
```
but in Job.Execute() method the logger's scope and someValue.Value are empty.

**Describe alternatives you've considered**
Is there any workaround for assign logger scope (or update AsyncLocal<> value) and this value will be applied to Job.Execute?
without modifying the Jobs code.

**Comment by lahma** [comment, marked as answer]
Jobs run in dedicated thread so AsyncLocal is not available. Data should be passed in via trigger/job data map.
